tests/system-broker/tests/auth_test.go

```diff
@@ -95,6 +115,66 @@ func TestSystemBrokerAuthentication(t *testing.T) {
 	})
 }
 
+func TestCallingORDServiceWithCert(t *testing.T) {
+	ORDContext, err := pkg.NewTestContext(config)
+	require.NoError(t, err)
+
+	logrus.Infof("registering runtime with name: %s, within tenant: %s", runtimeInput.Name, ORDContext.Tenant)
+	runtime := director.RegisterRuntimeFromInputWithinTenant(t, ORDContext.Context, ORDContext.DexGraphqlClient, ORDContext.Tenant, runtimeInput)
+	defer director.UnregisterRuntimeWithinTenant(t, ORDContext.Context, ORDContext.DexGraphqlClient, ORDContext.Tenant, runtime.ID)
+
+	app, err := director.RegisterApplicationWithinTenant(t, ORDContext.Context, ORDContext.DexGraphqlClient, ORDContext.Tenant, applicationInput)
+	require.NoError(t, err)
+	defer director.UnregisterApplication(t, ORDContext.Context, ORDContext.DexGraphqlClient, ORDContext.Tenant, app.ID)
+
+	securedClient, configuration, certChain := getSecuredClientByContext(t, testCtx, runtime.ID)
+
+	bundle := director.CreateBundle(t, ORDContext.Context, ORDContext.DexGraphqlClient, app.ID, ORDContext.Tenant, testBundleName)
+	api := director.AddAPIToBundleWithInput(t, ORDContext.Context, ORDContext.DexGraphqlClient, bundle.ID, ORDContext.Tenant, apiDefInput)
+
+	t.Run("Should succeed calling ORD service with the same cert used for calling catalog", func(t *testing.T) {
+		req := createCatalogRequest(t)
+		resp, err := securedClient.Do(req)
+
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		require.Equal(t, resp.StatusCode, http.StatusOK)
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(t, err)
+		require.Contains(t, string(body), "services")
+
+		reqORD := createORDServiceRequest(t, ORDContext, api.ID, api.Spec.ID)
+		respORD, err := securedClient.Do(reqORD)
+
+		require.NoError(t, err)
+		require.NotNil(t, respORD)
+		require.Equal(t, http.StatusOK, respORD.StatusCode)
+	})
+
+	t.Run("Should fail calling ORD service with revoked cert", func(t *testing.T) {
+
+		logrus.Infof("revoking cert for runtime with id: %s", runtime.ID)
+		connectorClient := connector.NewCertificateSecuredConnectorClient(*configuration.ManagementPlaneInfo.CertificateSecuredConnectorURL, testCtx.ClientKey, certChain...)
+		ok, err := connectorClient.RevokeCertificate()
+		require.NoError(t, err)
+		require.Equal(t, ok, true)
+
+		req := createORDServiceRequest(t, ORDContext, api.ID, api.Spec.ID)
+
+		resp, err := securedClient.Do(req)
+
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		require.Equal(t, http.StatusBadRequest, resp.StatusCode) //oathkeeper cannot be configured to return 401 the way we use it
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(t, err)
+		require.Contains(t, string(body), "invalid tenantID")
```

Comment:
invalid tenantID?

Reply:
When the certificate is not resolved successfully (bc it had been revoked) the call to ORD Service will contain an (invalid) token which does not have the tenantId claim so this message is [returned](https://github.com/kyma-incubator/ord-service/blob/722f1717632f358a052cd4da0c759804d35af190/components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/controller/SpecificationsController.java#L28).